internal/commands/chat-sast.go

```diff
@@ -117,23 +96,152 @@ func runChatSast(chatWrapper wrappers.ChatWrapper, tenantWrapper wrappers.Tenant
 	}
 }
 
-func isAiGuidedRemediationEnabled(tenantWrapper wrappers.TenantConfigurationWrapper) bool {
+func getSastConversationDetails(
+	cmd *cobra.Command,
+	chatConversationID string,
+	statefulWrapper wrapper.StatefulWrapper,
+) (isNewConversation bool, userInput string, conversationId uuid.UUID, err error, done bool) {
+	newConversation := false
+	if chatConversationID == "" {
+		newConversation = true
+		chatConversationID = statefulWrapper.GenerateId().String()
+	} else {
+		userInput, _ = cmd.Flags().GetString(params.ChatUserInput)
+		if userInput == "" {
+			msg := fmt.Sprintf(UserInputRequiredErrorFormat, params.ChatUserInput, params.ChatConversationID)
+			logger.PrintIfVerbose(msg)
+			return false, "", uuid.UUID{}, outputError(cmd, uuid.Nil, errors.Errorf(msg)), true
+		}
+	}
+
+	id, err := uuid.Parse(chatConversationID)
+	if err != nil {
+		logger.PrintIfVerbose(err.Error())
+		return false, "", uuid.UUID{}, outputError(cmd, id, errors.Errorf(ConversationIDErrorFormat, chatConversationID)), true
+	}
+	return newConversation, userInput, id, nil, false
+}
+
+func buildSastMessages(cmd *cobra.Command, newConversation bool, scanResultsFile, sastResultID, sourceDir string, id uuid.UUID, userInput string) ([]message.Message, error, bool) {
+	var newMessages []message.Message
+	if newConversation {
+		systemPrompt, userPrompt, e := sastchat.BuildPrompt(scanResultsFile, sastResultID, sourceDir)
+		if e != nil {
+			logger.PrintIfVerbose(e.Error())
+			return nil, outputError(cmd, id, e), true
+		}
+		newMessages = append(newMessages, message.Message{
+			Role:    role.System,
+			Content: systemPrompt,
+		}, message.Message{
+			Role:    role.User,
+			Content: userPrompt,
+		})
+	} else {
+		newMessages = append(newMessages, message.Message{
+			Role:    role.User,
+			Content: userInput,
+		})
+	}
+	return newMessages, nil, false
+}
+
+func CreateStatefulWrapper(cmd *cobra.Command, azureAiEnabled, checkmarxAiEnabled bool, tenantConfigurationResponses *[]*wrappers.TenantConfigurationResponse) (
+	statefulWrapper wrapper.StatefulWrapper, customerToken string) {
+	conn := connector.NewFileSystemConnector("")
+
+	customerToken, _ = wrappers.GetAccessToken()
+
+	if azureAiEnabled {
+		aiProxyAzureAIRoute := viper.GetString(params.AiProxyAzureAiRouteKey)
+		aiProxyEndPoint, _ := wrappers.GetURL(aiProxyAzureAIRoute, customerToken)
+		model, _ := GetAzureAiModel(tenantConfigurationResponses)
+		statefulWrapper, _ = wrapper.NewStatefulWrapperNew(conn, aiProxyEndPoint, customerToken, model, dropLen, 0)
+	} else if checkmarxAiEnabled {
+		aiProxyCheckmarxAIRoute := viper.GetString(params.AiProxyCheckmarxAiRouteKey)
+		aiProxyEndPoint, _ := wrappers.GetURL(aiProxyCheckmarxAIRoute, customerToken)
```

Comment:
look at main.go, you should had to chatWrapper constructor this paths as params and use it from the chatWrapper object like feilds